src/ios/CDVCamera.m

```diff
@@ -459,56 +470,99 @@ - (NSData*)processImage:(UIImage*)image info:(NSDictionary*)info options:(CDVPic
     return data;
 }
 
-- (void)picker:(PHPickerViewController *)picker didFinishPicking:(NSArray<PHPickerResult *> *)results API_AVAILABLE(ios(14)) {
+- (void)documentPicker:(UIDocumentPickerViewController *)controller
+didPickDocumentsAtURLs:(NSArray<NSURL *> *)urls; {
     __weak CDVCamera* weakSelf = self;
     
+    if (urls.count > 0) {
+        NSURL *fileURL = [urls firstObject];
+        
+        NSError *error;
+        NSData *data = [NSData dataWithContentsOfURL:fileURL options:NSDataReadingMappedIfSafe error:&error];
+        
+        if (error) {
+            [self sendErrorResultWithMessage];
+            return;
+        }
+        
+        // Create a UIImage from the data
+        UIImage *original = [[UIImage alloc] initWithData:data];
+        [self handleImageFromPicker:original withData:data];
+    } else {
+        [self sendErrorResultWithMessage];
+    }
+}
+
+- (void)documentPickerWasCancelled:(UIDocumentPickerViewController *)controller; {
+    [self sendErrorResultWithMessage];
+}
+
+- (void)sendErrorResultWithMessage {
```

Comment:
This does more than the naming implies. More like onEndSelectionWithoutImage. And use it also in other instances (e.g. insisde picker:didFinishPicking:)